### PR TITLE
rv32i: Convert PMP disable to a nop

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

We never set the lock bit for PMP (and if we did we can't unset it
anyway) so PMP won't apply to Tock only apps. In that case we never need
to disable it as it will be fully reconfigured before running the next
app.

### Testing Strategy

This pull request was tested by running an app on the HiFive1 revB and the QEMU CI.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make format`.
- [X] Fixed errors surfaced by `make clippy`.

### CODE REVIEW

File: arch/rv32i/src/pmp.rs [hunk 1]
```diff
@@ -177,183 +177,7 @@ impl kernel::mpu::MPU for PMPConfig {
...
-        }
-        //set first PMP to have permissions to entire space
-        csr::CSR.pmpaddr0.set(0xFFFF_FFFF);
-        //enable R W X fields
-        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::r0::SET);
-        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::w0::SET);
-        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::x0::SET);
-        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::a0::TOR)
-    }
+    fn disable_mpu(&self) {}
```
Comment (hudson-ayers) [inline arch/rv32i/src/pmp.rs:180]: This probably needs a comment explaining why we don't do need to do anything here

**Comment by alistair23** [comment]
There is no harm in disabling the PMP, this is just more efficient. We waste a lot of cycles writing all these CSRs. In saying that if someone has set the lock bits, it can not be disabled, so the function as is won't work.

https://github.com/tock/tock/pull/1867 is more focused on ePMP so I would like to keep that a littler separate. https://github.com/tock/tock/pull/1867 will make no difference to regular PMP.

`disable_mpu_for_kernel()` isn't really right either as with ePMP it will never be disabled. Maybe we should just remove the disable function? I know Tock doesn't like init functions, but maybe we can have an init/clean function to call at bootup that resets anything the prior stage has done (if possible).

**Comment by alphan** [comment]
#1860

**Comment by alistair23** [comment]
Replaced with: https://github.com/tock/tock/pull/1873

**Pull request closed without merging**